### PR TITLE
fix: improves the readme files and organize them into their respective folders

### PR DESCRIPTION
**Pull request by kroglid**
Adds more details about Pulumi and how to configure the config
Adds prerequisites to root README.md
Adds a step guide in root README.md with links to more detailed config guides
Moves configs to their respective folders

### CODE REVIEW

File: keycloak/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Keycloak
+
+Keycloak is used to authenticate users of the application.
+
+Found in the `keycloak` folder. The image is based on the official Keycloak image, but with additional build time configuration and a custom realm. The realm is defined in the `realm.json` file, and it contains several environment variables that are used to configure the realm on launch.
```
Comment (daniejoh) [inline keycloak/README.md]: > Found in the `keycloak` folder.

We already are in this folder.

File: frontend/README.md [hunk 2]
```diff
@@ -4,12 +4,41 @@ This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next
 
 ## Getting Started
 
-First, run the development server:
+When running locally the application expects a `.env.local` file in the root of the `frontend` folder. The file should contain the following variables:
+
+- `KEYCLOAK_ID` - The client ID of the Keycloak client
```
Comment (daniejoh) [inline frontend/README.md]: How can I find this?

Comment (daniejoh) [reply frontend/README.md]: And is there any default?

File: infrastructure/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- `cloudflare:zoneId` - The zone ID of the Cloudflare zone to use
+- `flood-frontend:domainName` - The domain name to use for the flood-frontend application.
+
+## Deploying the infrastructure
+
+Use the AWS CLI to authenticate with AWS, and then run `pulumi up` to deploy the infrastructure.
+
+## Tear down the infrastructure
+
+To tear down pulumi, use `pulumi destroy`.
```
Comment (daniejoh) [inline infrastructure/README.md]: We dont tear down pulumi. We tear down infrastructure _with_ pulumi.

**Review by daniejoh — CHANGES_REQUESTED** [review]
Some small changes :)

**Review by daniejoh — APPROVED** [review]

**Review by kstigen — APPROVED** [review]

**Pull request merged**